packages/react-native-reanimated/Common/cpp/reanimated/Fabric/ReanimatedCommitHook.cpp

```diff
@@ -41,7 +44,12 @@ void ReanimatedCommitHook::maybeInitializeLayoutAnimations(SurfaceId surfaceId)
             // shouldn't invoke it twice for the same surface
             return;
           }
-          shadowTree.getMountingCoordinator()->setMountingOverrideDelegate(strongThis->layoutAnimationsProxy_);
+          if constexpr (StaticFeatureFlags::getFlag("SHARED_ELEMENT_TRANSITIONS")) {
+            shadowTree.getMountingCoordinator()->setMountingOverrideDelegate(
+                strongThis->layoutAnimationsProxyExperimental_);
+          } else {
+            shadowTree.getMountingCoordinator()->setMountingOverrideDelegate(strongThis->layoutAnimationsProxyLegacy_);
+          }
```

Comment:
How about something like this instead? This way you avoid duplicating `shadowTree.getMountingCoordinator()->setMountingOverrideDelegate(...)` in your code. Sadly, there's 
```suggestion
          const auto &delegate = []{
            if constexpr (StaticFeatureFlags::getFlag("SHARED_ELEMENT_TRANSITIONS")) {
              return strongThis->layoutAnimationsProxyExperimental_;
            } else {
              return strongThis->layoutAnimationsProxyLegacy_;
            }
          }();
          shadowTree.getMountingCoordinator()->setMountingOverrideDelegate(delegate);
```

Reply:
I feel like this makes it more complicated than it actually is